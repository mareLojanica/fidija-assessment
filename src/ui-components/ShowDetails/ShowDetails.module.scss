.show-details {
  display: flex;
  gap: 50px;
  width: 100%;

  @media (max-width: 768px) {
    flex-direction: column;
    align-items: center;
    gap: 10px;
  }
  &__title__wrapper {
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }

  &__content {
    display: flex;
    flex-direction: column;
    align-items: start;
    padding-top: 3rem;
    gap: 2rem;
    width: 60%;
    @media (max-width: 768px) {
      width: 100%;
      gap: 0.1rem;
      padding-top: 0;
    }
  }

  &__rating {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 10px 0;
    gap: 20px;
  }

  &__title {
    font-size: 36px;
    font-weight: bold;
  }

  @media (max-width: 768px) {
    flex-direction: column;
  }
}
