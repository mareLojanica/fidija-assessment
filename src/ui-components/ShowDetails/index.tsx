import React from "react"
import styles from "./ShowDetails.module.scss"
import ShowImage from "../ShowImage"
import StarRating from "../../ui-components/StarRating"
import { ShowDetailsProps } from "../../types/ui-component.types"

const ShowDetails: React.FC<ShowDetailsProps> = ({
	image,
	name,
	rating,
	summary,
}) => {
	return (
		<div className={styles["show-details"]} data-testid="show-details">
			<ShowImage image={image} name={name} />
			<div className={styles["show-details__content"]}>
				<div className={styles["show-details__rating"]}>
					<StarRating rating={rating ?? 0} />
					<span>{rating ? `${rating / 2} / 5` : "N/A"}</span>
				</div>
				<div className={styles["show-details__title__wrapper"]}>
					<h2
						className={styles["show-details__title"]}
						role="heading"
					>
						{name}
					</h2>
					<p
						className={styles["show-details__summary"]}
						dangerouslySetInnerHTML={{
							__html: summary ?? "No summary available.",
						}}
					/>
				</div>
			</div>
		</div>
	)
}

export default ShowDetails
