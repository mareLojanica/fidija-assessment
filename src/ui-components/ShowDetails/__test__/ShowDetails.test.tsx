import React from "react";
import { render, screen } from "@testing-library/react";
import ShowDetails from "../";
import "@testing-library/jest-dom";

jest.mock("../../ShowImage", () =>
  jest.fn(({ image, name }) => (
    <div data-testid="show-image">
      <img src={image?.original || ""} alt={name} />
    </div>
  ))
);

jest.mock("../../StarRating", () =>
  jest.fn(({ rating }) => (
    <div data-testid="star-rating">{`Rating: ${rating} / 10`}</div>
  ))
);

describe("ShowDetails Component", () => {
  const mockShow = {
    image: {
      medium: "https://example.com/show-medium.jpg",
      original: "https://example.com/show.jpg",
    },
    name: "Breaking Bad",
    rating: 9,
    summary: "<p>A high school chemistry teacher turned meth kingpin.</p>",
  };

  test("renders show name", () => {
    render(<ShowDetails {...mockShow} />);

    expect(screen.getByText("Breaking Bad")).toBeInTheDocument();
  });

  test("renders show image using ShowImage component", () => {
    render(<ShowDetails {...mockShow} />);

    const img = screen.getByTestId("show-image").querySelector("img");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "https://example.com/show.jpg");
    expect(img).toHaveAttribute("alt", "Breaking Bad");
  });

  test("renders star rating correctly", () => {
    render(<ShowDetails {...mockShow} />);

    expect(screen.getByTestId("star-rating")).toHaveTextContent(
      "Rating: 9 / 10"
    );
  });

  test("handles missing rating properly", () => {
    render(<ShowDetails {...mockShow} rating={0} />);

    expect(screen.getByText("N/A")).toBeInTheDocument();
  });

  test("renders summary with HTML content", () => {
    render(<ShowDetails {...mockShow} />);

    expect(
      screen.getByText("A high school chemistry teacher turned meth kingpin.")
    ).toBeInTheDocument();
  });

  test("renders default summary when summary is missing", () => {
    render(<ShowDetails {...mockShow} summary={"Lorem Ipsum!"} />);

    expect(screen.getByText("Lorem Ipsum!")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ShowDetails {...mockShow} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
