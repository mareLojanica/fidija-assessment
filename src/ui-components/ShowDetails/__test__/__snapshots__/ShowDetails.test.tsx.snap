// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDetails Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="show-details"
    data-testid="show-details"
  >
    <div
      data-testid="show-image"
    >
      <img
        alt="Breaking Bad"
        src="https://example.com/show.jpg"
      />
    </div>
    <div
      class="show-details__content"
    >
      <div
        class="show-details__rating"
      >
        <div
          data-testid="star-rating"
        >
          Rating: 9 / 10
        </div>
        <span>
          4.5 / 5
        </span>
      </div>
      <div
        class="show-details__title__wrapper"
      >
        <h1
          class="show-details__title"
        >
          Breaking Bad
        </h1>
        <p
          class="show-details__summary"
        />
        <p>
          A high school chemistry teacher turned meth kingpin.
        </p>
        <p />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
